jerry-core/ecma/builtin-objects/ecma-builtins.cpp

```diff
@@ -291,43 +291,91 @@ ecma_builtin_try_to_instantiate_property (ecma_object_t *object_p, /**< object *
 {
   JERRY_ASSERT (ecma_get_object_is_builtin (object_p));
 
-  ecma_property_t *built_in_id_prop_p = ecma_get_internal_property (object_p,
-                                                                    ECMA_INTERNAL_PROPERTY_BUILT_IN_ID);
-  ecma_builtin_id_t builtin_id = (ecma_builtin_id_t) built_in_id_prop_p->u.internal_property.value;
+  const ecma_object_type_t type = ecma_get_object_type (object_p);
 
-  JERRY_ASSERT (ecma_builtin_is (object_p, builtin_id));
+  if (type == ECMA_OBJECT_TYPE_BUILT_IN_FUNCTION)
+  {
+    ecma_string_t* magic_string_length_p = ecma_get_magic_string (LIT_MAGIC_STRING_LENGTH);
 
-  switch (builtin_id)
+    bool is_length_property = ecma_compare_ecma_strings (string_p, magic_string_length_p);
+
+    ecma_deref_ecma_string (magic_string_length_p);
+
+    if (is_length_property)
+    {
+      /*
+       * Lazy instantiation of 'length' property
+       *
+       * Note:
+       *      We don't need to mark that the property was already lazy instantiated,
+       *      as it is non-configurable and so can't be deleted
+       */
+
+      ecma_property_t *desc_prop_p = ecma_get_internal_property (object_p,
+                                                               ECMA_INTERNAL_PROPERTY_BUILT_IN_ROUTINE_DESC);
```

Comment:
Current style checker doesn't check alignment inside braces, so the line was just skipped by the checker script.

Thanks for the note. I'll fix.